### PR TITLE
Updates Heterogeneous and Heavy Tailed data generators to use numpy.random.Generator

### PR DESCRIPTION
**Pull request by matthewclegg**

### CODE REVIEW

File: src/data_generators/tests/heavy_tailed_impression_generator_test.py [hunk 1]
```diff
@@ -23,10 +22,10 @@
 
 class ImpressionGeneratorTest(absltest.TestCase):
     def test_heterogeneous_impression_generator(self):
-        generator = HeavyTailedImpressionGenerator(5, 2, RandomState(0))
+        generator = HeavyTailedImpressionGenerator(5, 2)
         generated_ids = generator()
         self.assertEqual(set(generated_ids), set([0, 1, 2, 3, 4]))
-        self.assertLen(generated_ids, 33)
+        self.assertLen(generated_ids, 13)
```
Comment (riemanli) [inline src/data_generators/tests/heavy_tailed_impression_generator_test.py:28]: How do we get the number `13` here? Is it solely based on the result of seed==1?

File: src/data_generators/heterogeneous_impression_generator.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def __init__(
         n: int,
         gamma_shape: float = 1,
         gamma_scale: float = 1,
-        random_state: RandomState = None,
+        random_generator: np.random.Generator = None,
     ):
         """Constructor for the HeterogeneousImpressionGenerator.
```
Comment (riemanli) [inline src/data_generators/heterogeneous_impression_generator.py:35]: Not related to the changes. I think the term in exp() in the pdf of the Gamma distribution in the docstring should have a negative sign.

Comment (jiayu-google) [reply src/data_generators/heterogeneous_impression_generator.py:35]: Fixed in PR#37. Thanks Rieman.

File: src/data_generators/tests/heavy_tailed_impression_generator_test.py (hunk 1 above)
Comment (jiayu-google) [reply src/data_generators/tests/heavy_tailed_impression_generator_test.py:28]: Yes. Result changes when using a different seed.

(I wrote this test before and now realized that a test based on random seed is not a good practice. Let's keep it for now and I'll use "fake seed" for test in the future.)

**Review by jiayu-google — APPROVED** [review]
Thanks Matt!

**Review by riemanli — APPROVED** [review]
LGTM

**Review by mpatel24 — APPROVED** [review]

**Review by RonPinelliMRC — APPROVED** [review]
MRC approving based on cursory review after confirming non-independent code review on an exception basis. Experimental code will require independent review prior to publication of or reliance on results generated.

**Pull request merged**